Fixed CSS for publisher flipper

heyhippari opened the pull request:
# Fixed
- Fixed: Fixed the CSS for the publisher flipper

![image](https://github.com/user-attachments/assets/dd78afd0-a846-482d-802a-551880aba4c6)

As mentioned on Discord, one caveat is that the width will be based upon the first publisher.

heyhippari committed:
Fixed CSS for publisher flipper

majora2007 commented:
Just to let you know, I'm still planning to merge this in. Just not yet ready to do the testing to see if the width issue will be impactful.

majora2007 commented:
@MrRobotjs since you like CSS, could you look into this PR and see if you can remove the one constraint that's holding me from merging in (aka the width issue)?

MrRobotjs commented:
I tested this and images that have a longer width are still square. I think the proper fix would be to do what @therobbiedavis proposed here https://github.com/Kareadita/Kavita/issues/3451#issuecomment-2536613047 , i was already implementing this but I tried the idea of only showing the text and on hover a tooltip with the image will appear above instead. But I can also just make it so only the image appears if available or text if no image is available.

therobbiedavis committed:
Merge branch 'develop' into pr/3461

therobbiedavis committed:
Fixing publisher flipper

- Adjusted for dynamic publish width, but fixed height
- Adjusted divs for max-content
- Fixed flipper positioning

therobbiedavis commented:
@heyhippari @MrRobotjs @majora2007 I have fixed this up according to the comments.

Fesaa commented:
Visually, it looks good now. But there seems to be a bug in the flipper code, I only ever see it flipping through 2 or 3 publishers. I have 5 assigned now, but I can only see the first two, sometimes it seems to switch which publishers it flips through, but I don't know when

https://github.com/user-attachments/assets/49d48485-1262-4eaa-8868-f4a32cd9fac1

majora2007 commented:
Looks like there may be some leftover logic for me to handle. I can take care of the rest of this.

therobbiedavis committed:
copiliot feedback

therobbiedavis commented on UI/Web/src/app/_single-module/publisher-flipper/publisher-flipper.component.scss:
```diff
@@ -17,43 +17,47 @@
 
 .publisher-back {
-  transform: rotateX(180deg);
+  transform: rotateX(180deg) translateY(100%);
```
Ideally this would work, but because we needed the child to be relative for the max-content to work, this will not work properly

therobbiedavis commented on UI/Web/src/app/_single-module/publisher-flipper/publisher-flipper.component.scss:
```diff
@@ -17,43 +17,47 @@
+
+  // Position: absolute on the second child
+  &:not(:first-child) {
+    position: relative;
```
This is intentional to make the max-content work properly

therobbiedavis committed:
Merge branch 'develop' into pr/3461

majora2007 commented:
I still haven't forgotten about this FYI. I still want to bring this into Kavita.
